### PR TITLE
Feature/catalog endpoint

### PR DESCRIPTION
**Pull request by InfobyAdrienne**
Added ListCatalog to the catalog endpoint

### CODE REVIEW

File: api/handler/catalog.go [hunk 1]
```diff
@@ -47,3 +51,12 @@ func (h *CatalogHandler) CreateCatalog(ctx echo.Context) error {
 		Id: id.(string),
 	})
 }
+
+func (h *CatalogHandler) ListCatalog(ctx echo.Context) error {
```
Comment (onselakin) [inline api/handler/catalog.go:55]: Can you rename this to GetCatalog. We usually use the List methods when we want to return multiple resources (as in a Catalog list).

File: domain/plan.go [hunk 2]
```diff
@@ -153,8 +153,9 @@ func (p *Plan) JobSpecification() JobSpecification {
 
 	for _, task := range p.Tasks {
 		taskInfo := TaskInformation{
-			Id:    task.Id.Hex(),
-			Title: task.Title,
+			Id:       task.Id.Hex(),
```
Comment (onselakin) [inline domain/plan.go:156]: Why do we need these changes? If these changes were for fixing something else, could you perhaps try pulling the latest from the main?

File: go.mod [hunk 3]
```diff
@@ -15,6 +15,7 @@ require (
 
 require (
 	github.com/KyleBanks/depth v1.2.1 // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.3 // indirect
```
Comment (onselakin) [inline go.mod:18]: This seems out of place. Maybe it leaked in because of some other library?

File: go.mod [hunk 4]
```diff
@@ -38,12 +39,16 @@ require (
 	github.com/nats-io/nkeys v0.4.5 // indirect
 	github.com/nats-io/nuid v1.0.1 // indirect
 	github.com/rogpeppe/go-internal v1.11.0 // indirect
+	github.com/russross/blackfriday/v2 v2.1.0 // indirect
```
Comment (onselakin) [inline go.mod:42]: Same for this one

**Review by onselakin — CHANGES_REQUESTED** [review]
Looks good but has some unrelated dependencies in go.mod. Can you try with `go mod tidy` so it removes the unused ones.

File: domain/plan.go (hunk 2 above)
Comment (InfobyAdrienne) [reply domain/plan.go:156]: Yes I need to pull from the main

File: go.mod (hunk 4 above)
Comment (InfobyAdrienne) [reply go.mod:42]: I ran a go mod tidy

File: go.mod (hunk 3 above)
Comment (InfobyAdrienne) [reply go.mod:18]: `go mod tidy` fixed this

**Comment by radiosilence** [comment]
We need to add swag comment blocks to each endpoint so they are included in the swagger

**Pull request closed without merging**